test/cpp/interop/rpc_behavior_lb_policy.cc

```diff
@@ -96,19 +96,22 @@ class RpcBehaviorLbPolicy : public LoadBalancingPolicy {
   class Picker : public SubchannelPicker {
    public:
     Picker(RefCountedPtr<SubchannelPicker> delegate_picker,
-           RefCountedPtr<RpcBehaviorLbPolicy> parent)
-        : delegate_picker_(std::move(delegate_picker)), parent_(parent) {}
+           absl::string_view rpc_behavior)
+        : delegate_picker_(std::move(delegate_picker)),
+          rpc_behavior_(rpc_behavior) {}
 
     PickResult Pick(PickArgs args) override {
-      args.initial_metadata->Add(kRpcBehaviorMetadataKey,
-                                 parent_->rpc_behavior_);
+      char* rpc_behavior_copy = static_cast<char*>(
+          args.call_state->Alloc(rpc_behavior_.length() + 1));
+      strncpy(rpc_behavior_copy, rpc_behavior_.c_str(), rpc_behavior_.length());
```

Comment:
I think this should use `strcpy()` here, not `strncpy()`.  The latter has some edge cases that don't properly add the trailing nul character, and in this case, the bounds-checking isn't necessary anyway, because we're allocating the right amount of space before we do the copy.

Reply:
Done